apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -111,25 +116,111 @@
       case 'lockscreen-inputappclosed':
         this.updatePassCodeUI();
         break;
+      case 'touchstart':
+        // TODO: ignore while on error timeout
+        touch = evt.changedTouches[0];
+        target = document.elementFromPoint(touch.clientX, touch.clientY);
+        key = this._keyForTarget(target);
+        if (!key) {
+          break;
+        }
+        if (this.lastTouched) {
+          this._makeKeyInactive(this.lastTouched);
+        }
+        this._makeKeyActive(target);
+        this.lastTouched = target;
+        if (this.states.padVibrationEnabled &&
+          !this.states.passCodeErrorTimeoutPending) {
+          navigator.vibrate(this.configs.padVibrationDuration);
+        }
+        break;
+      case 'touchmove':
+        // TODO: ignore while on error timeout
+        touch = evt.changedTouches[0];
+        target = document.elementFromPoint(touch.clientX, touch.clientY);
```

Comment:
On touchmove, .target fails to report the current position under the finger. See comment above.